oauth2-useragent-core/src/main/java/com/microsoft/alm/oauth2/useragent/UserAgentImpl.java

```diff
@@ -84,6 +85,14 @@ AuthorizationResponse encode(final String methodName, final String... parameters
         provider.augmentProcessParameters(command, classPath);
         // TODO: is this the best way to add our JAR?
         classPath.add(System.getProperty("java.class.path"));
+
+        // Attempt to locate our JAR
+        final ArchiveLocator locator = new ArchiveLocator();
+        final String myClassPath = locator.locateArchive(UserAgentImpl.class);
+        if (myClassPath != null) {

```

Comment:
> The "java.class.path" value is a safety net, what if user passed the JavaFX jar in the classpath?

The `JavaFxProvider` already augments the `classPath` list based on probing the following locations:
```java
    private final File[] potentialJavaFxJarLocations = new File[]{
        new File(JAVA_HOME, "/lib/jfxrt.jar"),
        new File(JAVA_HOME, "/lib/ext/jfxrt.jar"),
    };
```